src/xrUICore/Lines/UILines.cpp

```diff
@@ -443,45 +443,45 @@ float CUILines::GetVIndentByAlign()
 // %c[255,255,255,255]
 u32 CUILines::GetColorFromText(const xr_string& str) const
 {
-    StrSize begin, end, comma1_pos, comma2_pos, comma3_pos;
+    StrSize begin = str.find(BEGIN);
+    const StrSize end = str.find(_END, begin);
 
-    begin = str.find(BEGIN);
-    end = str.find(_END, begin);
+    // Check if there even is a color tag
     if (begin == npos || end == npos)
         return m_dwTextColor;
-    // try default color
-    if (npos != str.find("%c[default]", begin, end - begin))
+
+    const xr_string color_tag = str.substr(begin, end - begin);
+
+    // Try default color
+    if (color_tag == "%c[default]")
         return m_dwTextColor;
 
     // Try predefined in XML colors
-    for (CUIXmlInitBase::ColorDefs::const_iterator it = CUIXmlInitBase::GetColorDefs()->begin();
-         it != CUIXmlInitBase::GetColorDefs()->end(); ++it)
+    const auto* color_defs = CUIXmlInitBase::GetColorDefs();
+    if (color_defs->contains(color_tag.c_str()))
```

Comment:
This is actually wrong.
ColorDefs are containing solely the color names, so it should be searching for `default` instead of `%c[default]`, like the new code does.

Reply:
Oh youre correct. Good catch :)